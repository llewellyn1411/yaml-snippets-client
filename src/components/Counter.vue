<template>
    <div class="counter-container">
        <div class="counter-value">{{ value }}</div>
        <div class="counter-label">{{ label }}</div>
    </div>
</template>

<script>
export default {
    name: 'Counter',
    components: {},
    props: {
        value: {
            type: Number,
            default: 0
        },
        label: {
            type: String,
            default: ''
        }
    }
};
</script>
<style lang="scss">
.counter-container {
    padding: 0 10px;
    box-sizing: border-box;
    text-align: right;

    .counter-value {
        font-weight: bold;
    }

    .counter-label {
        text-transform: capitalize;
    }
}
</style>
