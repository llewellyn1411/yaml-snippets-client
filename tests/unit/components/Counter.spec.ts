import { shallowMount } from '@vue/test-utils';
import Counter from '../../../src/components/Counter.vue';

describe( 'Counter.vue', () => {
    const wrapper = shallowMount( Counter, {
        propsData: {
            value: 987,
            label: 'things'
        }
    } );

    test( `renders correctly`, () => {
        expect( wrapper.html() ).toMatchSnapshot();
    } );

    test( 'should have a label', () => {
        const label = wrapper.find( '.counter-label' );
        expect( label.is( 'div' ) ).toBeTruthy();
        expect( label.text() ).toEqual( 'things' );
    } );

    test( 'should have a value', () => {
        const value = wrapper.find( '.counter-value' );
        expect( value.is( 'div' ) ).toBeTruthy();
        expect( value.text() ).toEqual( '987' );
    } );
} );
