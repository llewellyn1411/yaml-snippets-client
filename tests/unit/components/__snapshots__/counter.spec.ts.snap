// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue renders correctly 1`] = `
<div class="counter-container">
  <div class="counter-value">987</div>
  <div class="counter-label">things</div>
</div>
`;
